apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.py

```diff
@@ -62,14 +64,22 @@ def prepare(element_html: str, data: pl.QuestionData) -> None:
     )
     element_text = element_inner_html(element)
 
-    file_name = pl.get_string_attrib(element, "file-name")
+    file_name = pl.get_string_attrib(element, "file-name", None)
     if "_required_file_names" not in data["params"]:
         data["params"]["_required_file_names"] = []
+    # This is a special-case of the uniqueness test with a special message for the default file name.
+    elif (
+        file_name is None
+        and FILE_NAME_DEFAULT in data["params"]["_required_file_names"]
```

Comment:
It might be slightly simpler to check if the file name is redundant (i.e. fold into the check below), and then output a different error message if the file name is the default? Either that or explicitly check if the file name attribute is set with a boolean.

Reply:
Changed in 2686b626882cf5166cc5dc58f3f91ee3129c2924. Is this cleaner?